src/fonduer/parser/visual_linker.py

```diff
@@ -171,7 +174,7 @@ def _coordinates_from_HTML(
         pdf_word_list = sorted(
             pdf_word_list,
             key=lambda word_id__: block_coordinates[word_id__[0]]
-            + coordinate_map[word_id__[0]][1:3],
+            + (coordinate_map[word_id__[0]].top, coordinate_map[word_id__[0]].left),
```

Comment:
Can we sort them by all four coordinates? I am thinking there might be cases that you have one phrase being a subset of another phrase.

Reply:
Can you elaborate more on a case you are concerned about? Hopefully image with coordinates?

Reply:
For example, in the following special example, there might be two words `SMBT3904` and `SMBT3904...` that overlap with each other and in this case we want to use all four coordinates for sorting.
![Screen Shot 2020-06-15 at 4 08 37 PM](https://user-images.githubusercontent.com/5580008/84714507-b2b24d80-af22-11ea-95fc-24fd62bf3895.png)

Reply:
Thank you for the example.
Can we assume that the two overlapping words (ie `<word />`) are in the same `<line />` element?
If so, I think the two words are already sorted by pdftotext, meaning Fonduer does not have to sort them further.
Let me check if pdftotext sorts `<word />`. I may have to check the source code of it (eg https://gitlab.freedesktop.org/poppler/poppler/-/blob/master/utils/pdftotext.cc).

Reply:
Update: the `-layout` option of pdftotext "maintain original physical layout". With this option, I assume the `<word />` and other elements are sorted. I still have to check.

Reply:
Not quite sure, but pdftotext with `-layout` option sorts words by **top then left**.
https://gitlab.freedesktop.org/poppler/poppler/-/blob/master/poppler/TextOutputDev.cc#L2327

Reply:
Sound good! We can fix this later if this is an issue.